tests/npc_test.cpp

```diff
@@ -315,75 +316,105 @@ static void check_npc_movement( const tripoint &origin )
-
-    // Place an ambulance at the player's position, rotated correctly, 100% fuel, perfect condition
-    vehicle *veh = here.add_vehicle( vehicle_prototype_ambulance, player_pos, -90_degrees, 100, 2 );
-    // And make sure the game knows the player is in it so the NPC can follow
-    here.board_vehicle( player_pos, &pc );
-    REQUIRE( veh != nullptr );
-
-    // Create the NPC
     shared_ptr_fast<npc> guy = make_shared_fast<npc>();
     guy->normalize();
     guy->randomize();
-    // Spawn the NPC in the location desired
     guy->spawn_at_precise( tripoint_abs_ms( get_map().getabs( npc_pos ) ) );
     overmap_buffer.insert_npc( guy );
     g->load_npcs();
     guy->companion_mission_role_id.clear();
-    // Ensure they aren't trying to guard anything
     guy->guard_pos = std::nullopt;
-    // Reset the NPC, so they don't need to eat, or want to play with their items
     clear_character( *guy );
-    // But set them back to the desired position, because clear_character also sets position
     guy->setpos( npc_pos );
-
-    // Make the NPC a follower of the player
     talk_function::follow( *guy );
 
-    //And ensure the NPC has been placed at that location
-    npc *companion = get_creature_tracker().creature_at<npc>( npc_pos );
-    REQUIRE( companion != nullptr );
-
-    /* Uncomment for some extra info when test fails
-    debug_mode = true;
-    debugmode::enabled_filters.clear();
-    debugmode::enabled_filters.emplace_back( debugmode::DF_NPC );
-    REQUIRE( debugmode::enabled_filters.size() == 1 );
-    */
-
-    // Give the NPC 20 moves to board the vehicle
-    for( int i = 0; i < 20; ++i ) {
-        companion->moves = 100;
-        /* Uncommment for extra debug info
-            tripoint npc_pos = companion->pos();
-            optional_vpart_position vp = here.veh_at( npc_pos );
-            vehicle *veh = veh_pointer_or_null( vp );
-            add_msg( m_info, "%s is at %d %d %d and sees t: %s, f: %s, v: %s (%s)",
-                     companion->name, npc_pos.x, npc_pos.y, npc_pos.z,
-                     here.tername( npc_pos ),
-                     here.furnname( npc_pos ),
-                     vp ? remove_color_tags( vp->part_displayed()->part().name() ) : "",
-                     veh ? veh->name : " " );
-        */
-        companion->move();
-    }
+    return get_creature_tracker().creature_at<npc>( npc_pos );
+}
+
+TEST_CASE( "npc-board-player-vehicle" )
+{
+    REQUIRE_FALSE( test_data::npc_boarding_data.empty() );
+
+    for( std::pair<const std::string, npc_boarding_test_data> &given : test_data::npc_boarding_data ) {
+        GIVEN( given.first ) {
+            npc_boarding_test_data &data = given.second;
+            g->place_player( data.player_pos );
+            clear_map();
+            map &here = get_map();
+            Character &pc = get_player_character();
+
+            vehicle *veh = here.add_vehicle( data.veh_prototype, data.player_pos, 0_degrees, 100, 2 );
```

Comment:
The assumption that we want the player to be at the center point of the vehicle should be documented. This is true for the test vehicles added, but not all vehicles are constructed like this.

Reply:
The test is meant to use custom vehicles tailored to the test case. If we really want a vehicle with a different center than the player position, adding a separate vehicle spawn point is fairly trivial.